python-threatexchange/threatexchange/content_type/photo.py

```diff
@@ -19,3 +21,75 @@ class PhotoContent(ContentType):
       * frames from videos
       * thumbnails of videos
     """
+    @classmethod
+    def rotate_image(cls, image_data: bytes, angle: float) -> bytes: 
+        """
+        Rotate an image by a given angle 
+        """
+        with Image.open(io.BytesIO(image_data)) as img: 
+            rotated_img = img.rotate(angle, expand=True)
+            with io.BytesIO() as buffer: 
+                rotated_img.save(buffer, format=img.format)
+                return buffer.getvalue()
+
+    @classmethod
+    def flip_x(cls, image_data: bytes) -> bytes: 
+        """
+        Flip the image horizontally along the X-axis.
+        """
+        with Image.open(io.BytesIO(image_data)) as img: 
+            flipped_img = img.transpose(Image.FLIP_TOP_BOTTOM)
+            with io.BytesIO() as buffer: 
+                flipped_img.save(buffer, format=img.format)
+                return buffer.getvalue()
+
+    @classmethod
+    def flip_y(cls, image_data: bytes) -> bytes:
+        """
+        Flip the image vertically along the Y-axis.
+        """
+        with Image.open(io.BytesIO(image_data)) as img:
+            flipped_img = img.transpose(Image.FLIP_LEFT_RIGHT)
+            with io.BytesIO() as buffer:
+                flipped_img.save(buffer, format=img.format)
+                return buffer.getvalue()
+
+    @classmethod   
+    def flip_plus1(cls, image_data: bytes) -> bytes:
+        """
+        Flip the image diagonally along the line y = x.
+        """
+        with Image.open(io.BytesIO(image_data)) as img:
+            flipped_img = img.transpose(Image.Transpose.ROTATE_270).transpose(Image.FLIP_LEFT_RIGHT)
+            with io.BytesIO() as buffer:
+                flipped_img.save(buffer, format=img.format)
+                return buffer.getvalue()
+
+    @classmethod     
+    def flip_minus1(cls, image_data: bytes) -> bytes:
+        """
+        Flip the image diagonally along the line y = -x.
+        """
+        with Image.open(io.BytesIO(image_data)) as img:
+            flipped_img = img.transpose(Image.Transpose.ROTATE_90).transpose(Image.FLIP_LEFT_RIGHT)
+            with io.BytesIO() as buffer:
+                flipped_img.save(buffer, format=img.format)
+                return buffer.getvalue()
+    
+    @classmethod
+    def try_all_rotations(cls, image_data: bytes): 
+        """
+        Try all possible rotations to the image
```

Comment:
nit: Something like 
```
"""
Generate the 8 naive rotations of an image.

This can be helpful for testing, and for image algorithms that don't have
a native way to generate rotations during hashing, this can be a way to
brute force rotations.